### PR TITLE
Support for whitespace in mentions

### PR DESCRIPTION
**Pull request by justinneff**
## Checklist

- [x] Fix any eslint errors that occur
- [x] Update change-log for every plugin you touch
- [x] Add/Update tests if you add/change new functionality
- [x] Add/Update docs if you add/change functionality
- [x] Enable "Allow edits from maintainers" for this PR

## Use-case/Problem

When searching filtering mention suggestions with a list containing many people with same first name, the filter search value is cut off at the space. This adds an option that when enabled will allow the search value to contain the text between the trigger and cursor position to allow more refined filtering..

## Implementation

This is essentially the same fix as suggested in PR #1108 with the suggested fixes from the original PR applied and the change log updated.

There has been no meaningful activity on the original PR for some time leading me to believe has been abandoned. Therefore, I am opening my own PR in hopes of getting a resolution to the original issue #316.

## Demo

mentionwhitespacedemo

### CODE REVIEW

**Comment by juliankrispel** [comment]
Tested. Works fine for me! @justinneff can you please add a story then we're good to merge!

**Comment by justinneff** [comment]
@juliankrispel I am kind of new to contributing so I am not sure what you wanting for add a story. There is already issue #316 for this feature. Are you wanting me to create another issue?

**Comment by justinneff** [comment]
@juliankrispel I just pushed a commit to add the storybook. It was pretty straight forward. Just took me a few days to find time to take a look at it.

**Comment by justinneff** [comment]
Hey @tcosentino, it does appear that this also makes it so that you no longer need the leading space.

I am hopeful that this will be merged and included in a release of the official plugin very soon. I think I pushed the last piece that was needed before it can be merged tonight.

In the meantime, there are some notes in the [CONTRIBUTING.md](https://github.com/draft-js-plugins/draft-js-plugins/blob/master/CONTRIBUTING.md) file about referencing a fork in another project that may be helpful.

**Comment by juliankrispel** [comment]
just having a look at this

File: draft-js-mention-plugin/src/MentionSuggestions/__test__/mentionSuggestionStrategy.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  beforeEach(() => {
+    callback = sinon.spy();
+  });
+
+  context('when whitespace support is disabled', () => {
+    it('should match a word', () => {
+      nonWhitespaceStrategy(getBlock('@the'), callback);
+      expect(callback.callCount).to.equal(1);
+      expect(callback.lastCall.args).to.deep.equal([0, 4]);
+    });
```
Comment (juliankrispel) [inline draft-js-mention-plugin/src/MentionSuggestions/__test__/mentionSuggestionStrategy.js]: this is excellent!

**Comment by JustCodin** [comment]
@justinneff @juliankrispel Just wondering if either of you had a chance to take a peek at #1182.

**Comment by 1st8** [comment]
Thank you very much for this feature, you just saved me a quite some work. 🍻

**Comment by scottshuffler** [comment]
Does this still work? I'm not able to use whitespace currently using:

```
    "draft-js": "^0.11.1",
    "draft-js-emoji-plugin": "^2.1.3",
    "draft-js-focus-plugin": "^3.0.1",
    "draft-js-hashtag-plugin": "^2.0.4",
    "draft-js-inline-toolbar-plugin": "^3.0.1",
    "draft-js-linkify-plugin": "^2.0.2",
    "draft-js-mention-plugin": "^3.1.5",
    "draft-js-plugins-editor": "^3.0.0",
```

**Comment by shiranZe** [comment]
Thank you for this feature
but now,if the mention is in the middle of the sentence,the suggestions Menu opens only after clicking "space" key and not "@" (that's my configuration)

`createMentionPlugin({
        supportWhitespace: true,
        mentionPrefix: '@',`

and the result is:

captured (1)

anyone had this problem?

**Comment by fxOne** [comment]
This PR should fix the issue: https://github.com/draft-js-plugins/draft-js-plugins/pull/1507

**Comment by shiranZe** [comment]
@fxOne Thank you,
but now it when trying to delete the mention, only the first word deleted, instead of the whole block.

captured (2)

**Pull request merged**